File: RoQui/src/main/kotlin/dev/joguenco/roqui/note/delivery/model/DeliveryNoteReceiver.kt
package dev.joguenco.roqui.note.delivery.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.Date
import java.util.UUID
import org.hibernate.annotations.Immutable

/**
 * Un destinatario de la guia: el cliente de cada factura del viaje.
 *
 * El SRI admite varios por guia, y cada uno lleva su propio motivo de traslado y su documento de
 * sustento.
 */
@Entity
@Immutable
@Table(name = "v_ele_delivery_notes_receiver")
class DeliveryNoteReceiver {
    @Id val id: UUID? = null

    @Column(name = "code") val code: String? = null

    @Column(name = "number") val number: String? = null

    @Column(name = "line") val line: Long? = null

    @Column(name = "identification_type") val identificationType: String? = null

    @Column(name = "identification") val identification: String? = null

    @Column(name = "legal_name") val legalName: String? = null

    @Column(name = "address") val address: String? = null

    @Column(name = "transfer_reason") val transferReason: String? = null

    @Column(name = "code_document_support") val codeDocumentSupport: String? = null

    @Column(name = "number_document_support") val numberDocumentSupport: String? = null

    @Column(name = "authorization_document_support")
    val authorizationDocumentSupport: String? = null

    @Column(name = "date_document_support", columnDefinition = "DATE")
    val dateDocumentSupport: Date? = null
}

File: RoQuiApi/RoQui/Document/Model/DeliveryNote/DeliveryNoteReceiver.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace RoQuiApi.RoQui.Invoice.Model;

[Table("delivery_notes_receiver")]
public class DeliveryNoteReceiver
{
    [Column("id")]
    public int Id { get; set; }

    [Column("line", TypeName = "int")]
    public required int Line { get; set; }

    [Column("identification_type", TypeName = "varchar")]
    public required string IdentificationType { get; set; }

    [Column("identification", TypeName = "varchar")]
    public required string Identification { get; set; }

    [Column("legal_name", TypeName = "varchar")]
    public required string LegalName { get; set; }

    [Column("address", TypeName = "varchar")]
    public required string Address { get; set; }

    [Column("transfer_reason", TypeName = "varchar")]
    public required string TransferReason { get; set; }

    [Column("code_document_support", TypeName = "varchar")]
    public string? CodeDocumentSupport { get; set; }

    [Column("number_document_support", TypeName = "varchar")]
    public string? NumberDocumentSupport { get; set; }

    [Column("authorization_document_support", TypeName = "varchar")]
    public string? AuthorizationDocumentSupport { get; set; }

    [Column("date_document_support", TypeName = "date")]
    public DateTime? DateDocumentSupport { get; set; }

    [Column("delivery_note_id")]
    public int DeliveryNoteId { get; set; }

    public required DeliveryNote DeliveryNote { get; set; }

    public virtual required ICollection<DeliveryNoteReceiverDetail> DeliveryNoteReceiverDetails { get; set; }
}
